chemprop/cli/utils/parsing.py

```diff
@@ -147,10 +147,13 @@ def make_datapoints(
     weights = np.ones(N) if weights is None else weights
     gt_mask = [None] * N if gt_mask is None else gt_mask
     lt_mask = [None] * N if lt_mask is None else lt_mask
-    X_f = [None] * N if X_f is None else X_f
-    V_fs = [None] * N if V_fs is None else V_fs
-    E_fs = [None] * N if E_fs is None else E_fs
-    V_ds = [None] * N if V_ds is None else V_ds
+
+    n_mols = len(smiss)
+    n_components = len(smiss) + len(rxnss)
+    X_d = [[None] * N] * n_components if X_d is None else X_d
```

Comment:
```suggestion
    X_d = [None] * N if X_d is None else X_d
```
Below you have `[MoleculeDatapoint.from_smi(... x_d=X_d[i], ...) for i in range(N)]` which makes me think X_d should be a list of length N. This also matches with the idea that if you have multiple components, then you should concatenate all your individual X_d arrays (hstack) and attach that to the first dataset in the list of datasets you give to MulticomponentDataset.